java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -920,4 +1042,34 @@ private boolean endsWithReturn(StatementTree stmt) {
     }
     return false;
   }
+
+  /**
+   * Utility method. Checks whether the value associated to a given map and given key is a non-empty
+   * string.
+   *
+   * @param map - map corresponding to a method property
+   * @param key - key to check the corresponding value
+   * @return true if value is a non-empty string
+   */
+  private boolean checkValueStringExistsAndNotEmpty(Map<String, Object> map, String key) {
```

Comment:
If we change the return type of this method from `boolean` to `String`, and return `null` when `false` is being returned, we can avoid more calls to the map subsequently. This can be `getValueStringFromMap`.  
The callers can then be simplified.